### PR TITLE
upgrade to phpunit 6

### PR DESCRIPTION
**Pull request by sndsgd**
This upgrades phpunit to version 6, which requires dropping support for php 5.

@komapa thoughts on this? php 5.6 was eol'd at the beginning of the year.

### CODE REVIEW

**Comment by komapa** [comment]
PHP5 is still the most used PHP version right now so I think we should keep the latest 5.x version. So I would still keep 5.6 as supported by our client (aka, we should not use PHP 7 only syntax and features).

File: phpunit.xml [hunk 1]
```diff
@@ -1,15 +1,29 @@
...
         <whitelist>
-            <directory suffix=".php">lib</directory>
+            <directory>./lib</directory>
         </whitelist>
     </filter>
-    <logging>
-        <log type="coverage-html" target="./coverage" highlight="true"/>
-    </logging>
-</phpunit>
+</phpunit>
```
Comment (komapa) [inline phpunit.xml]: new line

**Comment by komapa** [comment]
I see why you did it `phpunit/phpunit 6.5.9 requires php ^7.0 -> your PHP version (5.6.32) does not satisfy that requirement.`

If there is no good reason to drop PHP 5.6, I would suggest we use PHPUnit 5 for the foreseeable future. Unless we  need some advanced PHPUnit testing (which I do not think we do here), there is no reason to let PHPUnit dictate the version the library can be used with.

**Comment by sndsgd** [comment]
IMO it would be nice to stop requiring changes to conform to any [php versions that are EOL](http://php.net/supported-versions.php), but if you think we need to keep it, I can just hold off on adding cool new things ™️.

**Comment by oligriffiths** [comment]
Also note, if we bump php versions shat should be co sideref a major version bump, people on php5 would continue to be able to instal this current release.

**Comment by komapa** [comment]
Well that is bias, I think I saw a different graph recently which confirms what I said above, I can find it some time today but the point remains, we do not want to not allow PHP5 to use the client for  testing purposes.

**Pull request closed without merging**